File: tests/Fortran/Fixed/0024-kernels-and-copy.f
      subroutine foo(a, b, n)
      implicit none
      integer :: a(:)
      integer :: b(:)
      integer :: n
      integer :: i

!$acc kernels copyin(a)
!$acc loop
      do i=1,n
        a(i) = i + 1
      enddo
!$acc end kernels

!$acc kernels copyin(a) copyout(b)
!$acc loop
      do i=1,n
        b(i) = a(i) + 1
      enddo
!$acc end kernels

!$acc kernels copyin(a) copyout(b)
!$acc loop
      do i=1,n
        a(i) = i + 1
      enddo
!$acc loop
      do i=1,n
        b(i) = a(i) + 1
      enddo
!$acc end kernels

!$acc kernels loop copyin(a) copyout(b)
      do i=1,n
        b(i) = a(i) + 1
      enddo
!$acc end kernels loop

      end subroutine foo

File: tests/C/0024-kernels-and-copy.c
int main (int argc, char*argv[])
{
	unsigned N = 1000;
	unsigned a[N],b[N];

	#pragma acc kernels copyout(a[0:N])
	{
		#pragma acc loop
		for (unsigned i = 0; i < N; ++i)
			a[i] = i+1;
	}
	#pragma acc kernels copyin(a[0:N]) copyout(b[0:N])
	{
		#pragma acc loop
		for (unsigned i = 0; i < N; ++i)
			b[i] = a[i] + 1;
	}
	#pragma acc kernels copyin(a[0:N]) copyout(b[0:N])
	{
		#pragma acc loop
		for (unsigned i = 0; i < N; ++i)
			a[i] = i + 1;
		#pragma acc loop
		for (unsigned i = 0; i < N; ++i)
			b[i] = a[i] + 1;
	}
	#pragma acc kernels loop copyin(a[0:N]) copyout(b[0:N])
	for (unsigned i = 0; i < N; ++i)
		b[i] = a[i] + 1;

	return 0;
}
